db/migrate/20230121151323_add_target_to_assessment.rb

```diff
@@ -0,0 +1,17 @@
+class AddTargetToAssessment < ActiveRecord::Migration[7.0]
+  def change
+    # Existierende Zeilen updaten!
+    add_column :assessments, :target, :json
```

Comment:
Könnte man nicht besser das Gruppenziel auch als Target erfassen, und da die `student_id` einfach leerzulassen? Das Handling der Zielwerte im analysis-view dürfte dadurch einfacher werden, weil die Zielwerte nicht aus zwei unterschiedlichen Quellen zusammengesucht werden müssen. Außerdem werden dann nicht quasi-identische Sachen an zwei unterschiedlichen Stellen gespeichert.

Reply:
Da gebe ich dir völlig Recht.
Entstanden ist es so weil es zunächst einfacher war. Jetzt mit den Schülerzielen würde es Sinn machen die Klassenziele auch so zu händeln, allerdings gibt es da (soweit ich sehe) ein Problem:

Die Schülerziele sind jeweils an Schüler gebunden die es wirklich gibt. Damit meine ich, dass sie im ActiveRecord und der Datenbank existieren. Würde ich die Klassenziele so speichern wollen müsste ich als studentID den Wert -1 übergeben und dann würde nach diesem Schüler gesucht werden, den es aber gar nicht gibt.
Es sei denn wir führen ihn ein. Aber dann haben wir einen Schüler in der DB der anders ist als alle anderen... Das klingt irgendwie nicht sehr reizvoll.

Vielleicht ist es wirklich die bessere Lösung wenn man es sauber hinbekommt, aber um die Risiken für das allgemeine Funktionieren von Levumi bei meinen Änderungen gering zu halten habe ich es erstmal noch nicht so probiert.

Reply:
Kannst du nicht die `student_id` einfach leer lassen (`ǹull`)? Wenn ich das richtig sehe, musst du dafür im Model nur ein `optional: true` bei der `student`-Relation einbauen.

Reply:
Hah, gute Idee. Damit könnte es klappen. Hatte ganz ausgeblendet, dass man das in Rails ja auch machen kann.

Reply:
geändert in [8b6b241](https://github.com/muehling/levumi/pull/29/commits/8b6b2418a9e9f003763c5e80fdf7cb8ca564dff7)

Das hat den Code tatsächlich nochmal etwas vereinfacht, da so einige Funktionen einfach komplett weggefallen sind und das Speichern und Laden natürlich auch einfacher wurde.

Damit ist analysis-view.vue sogar wieder unter 1000 Zeilen :stuck_out_tongue_winking_eye: